### PR TITLE
integration-test (ticdc): providing option of downloading public accessible version of binaries for integration test

### LINKED ISSUE #8886: Setup integration test environment using public accessible binaries
### Is your feature request related to a problem?

At present, setting up the integration test environment requires building the Docker image and downloading associated binaries from an internal file server that is inaccessible from public networks. This poses a challenge for users without internal network access, making the process inconvenient for them.

### Describe the feature you'd like

Make it possible to build and run integration tests without requiring access to the PingCAP internal network.

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by charleszheng44**
### What problem does this PR solve?

providing option of downloading public accessible version of binaries for integration test

Issue Number: close #8886 

### What is changed and how it works?

Users can using public accessible binaries for the integration test by setting env COMMUNITY=true 

For example:

`COMMUNITY="true" CASE="clustered_index" make mysql_docker_integration_test_with_build`
`COMMUNITY="true" CASE="clustered_index" make kafka_docker_integration_test_with_build`

#### Tests 

 - Unit test
 - Integration test
 - [x] Manual test (add detailed scripts or steps below)
 
run ``COMMUNITY="true" CASE="clustered_index" make kafka_docker_integration_test_with_build`, and the specified integration test case will be run in docker images with public accessible binaries 

 - No code

##### Will it cause performance regression or break compatibility?
NO

##### Do you need to update user documentation, design documentation or monitoring documentation?
NO

### Release note 

```release-note
 `None`
```

### CODE REVIEW

File: scripts/download-integration-test-binaries.sh [hunk 1]
```diff
@@ -38,31 +45,113 @@ function download() {
...
+	local dist="${ver}-${os}-${arch}"
+	local tidb_file_name="tidb-community-server-$dist"
+	local tidb_tar_name="${tidb_file_name}.tar.gz"
+	local tidb_url="https://download.pingcap.org/$tidb_tar_name"
+	local toolkit_file_name="tidb-community-toolkit-$dist"
+	local toolkit_tar_name="${toolkit_file_name}.tar.gz"
+	local toolkit_url="https://download.pingcap.org/$toolkit_tar_name"
+
+	color-green "Download community binaries..."
+	download "$tidb_url" "$tidb_tar_name" "tmp/$tidb_tar_name"
```
Comment (hi-rustin) [inline scripts/download-integration-test-binaries.sh:68]: Which version it will download?

**Review by hi-rustin — COMMENTED** [review]
Thanks for adding it!

File: scripts/download-integration-test-binaries.sh (hunk 1 above)
Comment (charleszheng44) [reply scripts/download-integration-test-binaries.sh:68]: for now, 6.5.2 will be downloaded.

Comment (charleszheng44) [reply scripts/download-integration-test-binaries.sh:66]: The latest update includes a new commit that enables users to specify the version (default: 6.5.2), operating system (default: Linux), and architecture (default: amd64).

Comment (hi-rustin) [reply scripts/download-integration-test-binaries.sh:66]: Does the `version` conflict with the original `branch` parameter?

**Review by hi-rustin — APPROVED** [review]
Great work!

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**